pkgs/development/libraries/librsvg/default.nix

```diff
@@ -20,14 +24,26 @@ stdenv.mkDerivation rec {
 
   propagatedBuildInputs = [ glib gdk-pixbuf cairo ];
 
-  nativeBuildInputs = [ pkgconfig rustc cargo vala gobject-introspection ]
+  preConfigure = stdenv.lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+    mkdir -p .cargo
+    cat >> .cargo/config <<'EOF'
+    [target."${toRustTarget stdenv.buildPlatform}"]
+    "linker" = "${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc"
+    [target."${toRustTarget stdenv.hostPlatform}"]
+    "linker" = "${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc"
+    EOF
+  '';
```

Comment:
Can we share this with the other places we do this?

Reply:
Some setup hook would be nice if it is not possible to hard code this during compilation.